### PR TITLE
iOS input improvements

### PR DESCRIPTION
**Pull request by adly-holler**
- Add return action handler
- Add properties `keyboardType`, `returnKeyType`, `autocapitalizationType`, and `autocorrectionType`

### CODE REVIEW

File: FewDemo-iOS/ViewController.swift [hunk 1]
```diff
@@ -48,9 +48,9 @@ private func renderRow(row: Int) -> Element {
 }
 
 func renderTableView(component: Component<()>, state: ()) -> Element {
-	return TableView((1...100).map(renderRow), selectionChanged: println)
-		.flex(1)
-		.selfAlignment(.Stretch)
+    return TableView((1...100).map(renderRow), selectionChanged: println)
+        .flex(1)
+        .selfAlignment(.Stretch)
```
Comment (robrix) [inline FewDemo-iOS/ViewController.swift]: I think these were probably tabs on purpose.

File: Few-iOS/Input.swift [hunk 2]
```diff
@@ -43,19 +63,19 @@ public class Input: Element {
...
             if let oldInput = old as? Input {
-                let newTrampoline = oldInput.trampoline
-                newTrampoline.action = trampoline.action // Make sure the newest action is used
-                trampoline = newTrampoline
+                let newTrampoline = oldInput.actionTrampoline
+                newTrampoline.action = actionTrampoline.action // Make sure the newest action is used
+                actionTrampoline = newTrampoline
             }
             
+            textField.delegate = self.inputDelegate
```
Comment (joshaber) [inline Few-iOS/Input.swift]: Style: can drop the `self` here.

File: Few-iOS/Input.swift [hunk 3]
```diff
@@ -43,19 +63,19 @@ public class Input: Element {
...
+            
             if placeholder != textField.placeholder {
                 textField.placeholder = placeholder
             }
             
-            if let text = text {
-                if text != textField.text {
-                    textField.text = text
-                }
+            if let text = text where text != textField.text {
```
Comment (joshaber) [inline Few-iOS/Input.swift]: So good :cookie:

**Comment by joshaber** [comment]
This is fantastic. Just a style nit and a merge needed.

**Comment by adly-holler** [comment]
Hmmm right, moving over to tabs kind of ruined the diff. Want me to revert it and we'll tab it out later?

**Comment by joshaber** [comment]
Eh, whatever's easiest. Currently can't be merged cleanly tho :cry:

**Comment by adly-holler** [comment]
This one's ready

**Comment by joshaber** [comment]
:cocktail:

**Pull request merged**